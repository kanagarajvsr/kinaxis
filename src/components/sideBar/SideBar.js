import { Link } from 'react-router-dom';
import "./sideBar.scss";
import DashboardIcon from '@mui/icons-material/Dashboard';
import GroupIcon from '@mui/icons-material/Group';
import InfoIcon from '@mui/icons-material/Info';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const SideBar = () => {
    return (
        <div className='sidebar'>
            <div className='top'>
                <span className='logo'>Kinaxis</span>
            </div>
            <hr />
            <div className='center'>
                <ul>
                    <li>
                        <Link to="/dashboard" style={{ textDecoration: "none" }}>
                            <DashboardIcon className="icon"/>
                            <span>Dashboard </span>
                        </Link>
                    </li>
                    <li>
                        <Link to="/users" style={{ textDecoration: "none" }}>
                            <GroupIcon className="icon"/><span>Users </span>
                        </Link>  
                    </li>
                    <li>
                        <Link to="/orders" style={{ textDecoration: "none" }}>
                            <ShoppingCartIcon className="icon"/><span>Orders </span>
                        </Link>  
                    </li>
                    <li>
                        <Link to="/about" style={{ textDecoration: "none" }}>
                            <InfoIcon className="icon"/><span>About</span>
                        </Link>  
                    </li>
                </ul>
            </div> 

        </div>
    )
}

export default SideBar