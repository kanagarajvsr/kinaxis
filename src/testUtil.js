
import { BrowserRouter } from 'react-router-dom'
import { render,screen,fireEvent } from '@testing-library/react';
 
export const renderWithRouter = (ui, {route = '/'} = {}) => {
    window.history.pushState({}, 'Test page', route);
    const renders = render(ui, {wrapper: BrowserRouter})
    return { 
        screen,
        fireEvent,
        ...renders
    }
  }