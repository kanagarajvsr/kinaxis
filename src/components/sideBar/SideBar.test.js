import SideBar from './SideBar';
import {renderWithRouter} from '../../testUtil';
 
describe('SideBar render', () => {
 
  test('Update Snapshot', () => { 
      const utils =  renderWithRouter(<SideBar />)
      const {asFragment}=utils;
      expect(asFragment()).toMatchSnapshot()
    })
  
  });
   


