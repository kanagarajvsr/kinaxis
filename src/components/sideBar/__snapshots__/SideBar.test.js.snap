// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar render Update Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sidebar"
  >
    <div
      class="top"
    >
      <span
        class="logo"
      >
        Kinaxis
      </span>
    </div>
    <hr />
    <div
      class="center"
    >
      <ul>
        <li>
          <a
            href="/dashboard"
            style="text-decoration: none;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="DashboardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
              />
            </svg>
            <span>
              Dashboard 
            </span>
          </a>
        </li>
        <li>
          <a
            href="/users"
            style="text-decoration: none;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="GroupIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
              />
            </svg>
            <span>
              Users 
            </span>
          </a>
        </li>
        <li>
          <a
            href="/orders"
            style="text-decoration: none;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <span>
              Orders 
            </span>
          </a>
        </li>
        <li>
          <a
            href="/about"
            style="text-decoration: none;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="InfoIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
            <span>
              About
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
